Library/Services/Implementation/NfieldSurveyInvitationTemplatesService.cs

```diff
@@ -39,6 +51,40 @@ public Task<IEnumerable<InvitationTemplateModel>> GetAsync(string surveyId)
                 .FlattenExceptions();
         }
 
+        public Task<InvitationTemplateModel> UpdateAsync(string surveyId, InvitationTemplateModel invitationTemplate)
+        {
+            CheckRequiredStringArgument(surveyId, nameof(surveyId));
+            CheckRequiredArgument(invitationTemplate, nameof(invitationTemplate));
+
+            var uri = SurveyInvitationTemplatesUrl(surveyId);
+
+            var updatedInvitationTemplate = new UpdateInvitationTemplate
+            {
+                Body = invitationTemplate.Body,
+                Subject = invitationTemplate.Subject,
+                Name = invitationTemplate.Name,
+                InvitationType = invitationTemplate.InvitationType
+            };
+
+            return Client.PatchAsJsonAsync(uri + "/" + invitationTemplate.Id, updatedInvitationTemplate)
+                .ContinueWith(responseMessageTask => responseMessageTask.Result.Content.ReadAsStringAsync().Result)
+                .ContinueWith(stringTask => JsonConvert.DeserializeObject<InvitationTemplateModel>(stringTask.Result))
+                .FlattenExceptions();
+        }
+
+        public Task<bool> RemoveAsync(string surveyId, InvitationTemplateModel invitationTemplate)
+        {
+            CheckRequiredStringArgument(surveyId, nameof(surveyId));
+            CheckRequiredArgument(invitationTemplate, nameof(invitationTemplate));
+
+            var uri = SurveyInvitationTemplatesUrl(surveyId);
+
+            return Client.DeleteAsync(uri + "/" + invitationTemplate.Id)
+                .ContinueWith(response => response.Result.Content.ReadAsStringAsync().Result)
+                .ContinueWith(stringTask => JsonConvert.DeserializeObject<bool>(stringTask.Result))
```

Comment:
The delete from the api will return an object, you can check it on my PR 
https://github.com/NIPOSoftwareBV/nfield-source/pull/1100/files